modules/rest/src/test/java/org/apache/ignite/internal/rest/auth/TestAuthSubscriber.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.rest.auth;
+
+import io.micronaut.security.authentication.AuthenticationResponse;
+import org.jetbrains.annotations.Nullable;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+/** Implementation of {@link Subscriber} for tests. */
+public class TestAuthSubscriber implements Subscriber<AuthenticationResponse> {
+    private Subscription source;
+    private @Nullable AuthenticationResponse lastResponse;
+    private @Nullable Throwable lastError;
+
+    @Override
+    public void onSubscribe(Subscription subscription) {
+        source = subscription;
+        source.request(1);
+    }
+
+    @Override
+    public void onNext(AuthenticationResponse authenticationResponse) {
+        lastResponse = authenticationResponse;
+    }
+
+    @Override
+    public void onError(Throwable throwable) {
+        lastError = throwable;
+    }
+
+    @Override
+    public void onComplete() {
+        source.cancel();
+    }
+
+    public void reset() {
```

Comment:
why do you need this method? Can you simply create new subscribers if needed?

Reply:
removed the method